### PR TITLE
Aniorb Presence

### PR DESCRIPTION
**Pull request by ICEPrey**
detailsdisco
details
genresdisco
genres
mylistdisco
mylist
populardisco
popular
recentdisco
recent
watchdisco
watch

### CODE REVIEW

File: websites/A/Aniorb/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "author": {
+      "name": "ICEPrey",
+      "id": "256973318588989440"
+    },
+    "contributors": [
+      {
```
Comment (StrikerFRFX) [inline websites/A/Aniorb/dist/metadata.json:8]: Don't add yourself as contributor, remove the entire contributor field.

File: websites/A/Aniorb/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        "name": "ICEPrey",
+        "id": "256973318588989440"
+      }
+    ],
+    "service": "Aniorb",
+    "description": {
+      "en": "Aniorb is a free anime website with no ads and the highest quality animes."
+    },
+    "url": "aniorb.me",
+    "version": "1.0.1",
```
Comment (StrikerFRFX) [inline websites/A/Aniorb/dist/metadata.json:18]: 1.0.0 on release

File: websites/A/Aniorb/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    "logo": "https://i.imgur.com/ng8qXxu.png",
```
Comment (StrikerFRFX) [inline websites/A/Aniorb/dist/metadata.json:19]: Higher res image, use an upscaler? or search around...

File: websites/A/Aniorb/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    "thumbnail": "https://i.imgur.com/WCgM5dd.png",
```
Comment (StrikerFRFX) [inline websites/A/Aniorb/dist/metadata.json:20]: No blur on thumbnail, just screenshot the website homepage.

File: websites/A/Aniorb/dist/metadata.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    "color": "#000000",
+    "tags": ["aniorb", "anime"],
```
Comment (StrikerFRFX) [inline websites/A/Aniorb/dist/metadata.json:22]: More tags could be applied, check Kickassanime for reference.

File: websites/A/Aniorb/presence.ts [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  browsing: "presence.activity.browsing"
+});
+let video = {
+video: false,
+duration: 0,
+currentTime: 0,
+paused: true
+};
+
+/**
```
Comment (StrikerFRFX) [inline websites/A/Aniorb/presence.ts:16]: Remove the get Timestamps function entirely and its comments and replace instances with `presence.getTimestamps`

**Review by StrikerFRFX — CHANGES_REQUESTED** [review]
See changes also wtf is indentation on `presence.ts`? use Prettier from the vscode extension store lol

File: websites/A/Aniorb/presence.ts [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+let video = {
+video: false,
+duration: 0,
+currentTime: 0,
+paused: true
+};
+  const timestamps = presence.getTimestamps(
+    Math.floor(video.currentTime),
+    Math.floor(video.duration)
+  );
```
Comment (SlowLife1661) [inline websites/A/Aniorb/presence.ts:18]: *It's const*

File: websites/A/Aniorb/presence.ts [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+  document.querySelector("video") !== null &&
+  video.video
+) if (video !== null && !isNaN(video.duration)) {
+  const series =   document.querySelector("#__next > div > div.flex.justify-between > div.w-full.justify-center.items-center.min-h-screen.lg\\:h-full.lg\\:w-10\\/12 > div > div.flex.flex-col.pb-2.xl\\:w-player.justify-between.items-center.w-full.text-white.my-4 > div.w-full.py-4.uppercase.flex.flex-col.items-start.lg\\:items-start > span"),
+    seriesTitle = series.textContent.toUpperCase(),
+    episode = document.querySelector("#__next > div > div.flex.justify-between > div.w-full.justify-center.items-center.min-h-screen.lg\\:h-full.lg\\:w-10\\/12 > div > div.flex.flex-col.pb-2.xl\\:w-player.justify-between.items-center.w-full.text-white.my-4 > div.flex.w-full.justify-between.items-end > span").textContent,
+    timestamps = presence.getTimestamps(
+      Math.floor(video.currentTime),
+      Math.floor(video.duration)
+    );
```
Comment (SlowLife1661) [inline websites/A/Aniorb/presence.ts:74]: See above  ( Can't reassign the const )

**Review by SlowLife1661 — COMMENTED** [review]
Kek

File: websites/A/Aniorb/presence.ts (hunk 8 above)
Comment (SlowLife1661) [reply websites/A/Aniorb/presence.ts:74]: Also fix deepscan issues xd

File: websites/A/Aniorb/presence.ts (hunk 7 above)
Comment (ICEPrey) [reply websites/A/Aniorb/presence.ts:18]: Wait its not suppose to be?

Comment (SlowLife1661) [reply websites/A/Aniorb/presence.ts:18]: See below, and tbh I don't see the point why you define it here

File: websites/A/Aniorb/presence.ts (hunk 8 above)
Comment (ICEPrey) [reply websites/A/Aniorb/presence.ts:74]: Ohh i defined it twice wtf ty

File: websites/A/Aniorb/presence.ts [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+} else if (document.location.pathname.includes("/myList")) {
+  presenceData.details = "Browsing through";
+  presenceData.state = "My List";
+  presenceData.smallImageKey = "heart";
+} else if (document.location.pathname.includes("/details")) {
+  presenceData.details = "Browsing through";
+  presenceData.state = "Anime Details";
+  presenceData.smallImageKey = "reading";
+}
+else if (document.querySelector("#__next > div > div.flex.justify-between > div.w-full.justify-center.items-center.min-h-screen.lg\\:h-full.lg\\:w-10\\/12 > div > div.flex.flex-col.pb-2.xl\\:w-player.justify-between.items-center.w-full.text-white.my-4 > div.w-full.py-4.uppercase.flex.flex-col.items-start.lg\\:items-start > span").textContent.toUpperCase() !== null) {
```
Comment (SlowLife1661) [inline websites/A/Aniorb/presence.ts:58]: Declare this query somewhere maybe or just make it shorter pretty sure you can do that

File: websites/A/Aniorb/dist/metadata.json [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    "service": "Aniorb",
+    "description": {
+      "en": "Aniorb is a free anime website with no ads and the highest quality animes."
+    },
+    "url": "aniorb.me",
+    "version": "1.0.0",
+    "logo": "https://i.imgur.com/9qhIuod.png",
+    "thumbnail": "https://i.imgur.com/THrtHoe.png",
+    "color": "#000000",
+    "tags": ["aniorb", "ani", "orb", "4anime", "animelist", "aniwatch", "animekisa", "gogoanime", "anime" , "watching" , "popular", "streaming", "free", "fast", "1080p", "website"],
```
Comment (Bas950) [inline websites/A/Aniorb/dist/metadata.json:16]: why tags like 4anime animelist aniwatch animekisa etc.?

File: websites/A/Aniorb/dist/metadata.json [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    "category": "anime",
+    "iframe": true,
+    "readLogs": false
+  }
```
Comment (Bas950) [inline websites/A/Aniorb/dist/metadata.json:20]: just dont set it, its default to false

File: websites/A/Aniorb/presence.ts (hunk 7 above)
Comment (Bas950) [inline websites/A/Aniorb/presence.ts:18]: I bet this isnt working rn

**Review by Bas950 — CHANGES_REQUESTED** [review]
Fix codacy errors

File: websites/A/Aniorb/dist/metadata.json (hunk 10 above)
Comment (ICEPrey) [reply websites/A/Aniorb/dist/metadata.json:16]: better search?

File: websites/A/Aniorb/presence.ts (hunk 7 above)
Comment (ICEPrey) [reply websites/A/Aniorb/presence.ts:18]: Its not lmfao how do I fix it?

File: websites/A/Aniorb/dist/metadata.json (hunk 10 above)
Comment (Bas950) [reply websites/A/Aniorb/dist/metadata.json:16]: they need to be tags related to this service not to other services :/

**Review by Bas950 — CHANGES_REQUESTED** [review]
Fix codacy errors

**Comment by ICEPrey** [comment]
How do I fix this fuckery, please

File: websites/A/Aniorb/.eslintrc.js [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+    //extends: 'airbnb',
+    parser: 'babel-eslint',
+    env: {
+      browser: true
+    },
+    plugins: ['react'],
+      extends: ['eslint:recommended'],
+    rules: {
+  }
+    }
```
Comment (Bas950) [inline websites/A/Aniorb/.eslintrc.js:11]: remove

File: websites/A/Aniorb/presence.ts [hunk 13]
```diff
@@ -0,0 +1,102 @@
+/* eslint no-dupe-else-if: "error" */
+/*eslint linebreak-style: ["error", "windows"]*/
```
Comment (Bas950) [inline websites/A/Aniorb/presence.ts:2]: remove

**Review by Bas950 — CHANGES_REQUESTED** [review]
To fix linebreak: 
image

**Comment by ICEPrey** [comment]
> To fix linebreak:
> image

I do have it on LF i still get the same errors (on github not vscode)

File: websites/A/Aniorb/presence.ts [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+    delete presenceData.endTimestamp;
+  }
+  presence.setActivity(presenceData, !video.paused);
+}
+presenceData.buttons = [
+  {
+    label: "Watch Episode",
+    url: document.baseURI
+  }
+];
```
Comment (Bas950) [inline websites/A/Aniorb/presence.ts:91]: this is outside of the if case from the video??!

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/A/Aniorb/presence.ts (hunk 14 above)
Comment (ICEPrey) [reply websites/A/Aniorb/presence.ts]: fixed (read disco please)

File: websites/A/Aniorb/dist/metadata.json [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+    "description": {
+      "en": "Aniorb is a free anime website with no ads and the highest quality animes."
+    },
+    "url": "aniorb.me",
+    "version": "1.0.0",
+    "logo": "https://i.imgur.com/9qhIuod.png",
+    "thumbnail": "https://i.imgur.com/THrtHoe.png",
+    "color": "#000000",
+    "tags": ["aniorb", "ani", "orb"],
+    "category": "anime",
```
Comment (SlowLife1661) [inline websites/A/Aniorb/dist/metadata.json:17]: Not exist

**Review by SlowLife1661 — CHANGES_REQUESTED** [review]

File: websites/A/Aniorb/dist/metadata.json (hunk 15 above)
Comment (SlowLife1661) [reply websites/A/Aniorb/dist/metadata.json:17]: My bad 🤣

Comment (SlowLife1661) [reply websites/A/Aniorb/dist/metadata.json:17]: *I'm going to sleep now*

Comment (ICEPrey) [reply websites/A/Aniorb/dist/metadata.json:17]: all good g :))

**Review by ICEPrey — COMMENTED** [review]
fixed

File: websites/A/Aniorb/presence.ts [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+  document.querySelector("video") !== null &&
+  video.video
+) if (video !== null && !isNaN(video.duration)) {
+  const series =   document.querySelector("#__next > div > div.flex.justify-between > div.w-full.justify-center.items-center.min-h-screen.lg\\:h-full.lg\\:w-10\\/12 > div > div.flex.flex-col.pb-2.xl\\:w-player.justify-between.items-center.w-full.text-white.my-4 > div.w-full.py-4.uppercase.flex.flex-col.items-start.lg\\:items-start > span"),
+    seriesTitle = series.textContent.toUpperCase(),
+    episode = document.querySelector("#__next > div > div.flex.justify-between > div.w-full.justify-center.items-center.min-h-screen.lg\\:h-full.lg\\:w-10\\/12 > div > div.flex.flex-col.pb-2.xl\\:w-player.justify-between.items-center.w-full.text-white.my-4 > div.flex.w-full.justify-between.items-end > span").textContent,
+    timestamps = presence.getTimestamps(
+      Math.floor(video.currentTime),
+      Math.floor(video.duration)
+    );
```
Comment (Bas950) [inline websites/A/Aniorb/presence.ts:70]: you can use presence.getTimestampsfromMedia

File: websites/A/Aniorb/presence.ts [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+  presenceData.smallImageKey = video.paused ? "pause" : "play";
+  presenceData.smallImageText = video.paused
+    ? (await strings).pause
+    : (await strings).play;
+  presenceData.startTimestamp = timestamps[0];
+  presenceData.endTimestamp = timestamps[1];
+  presence.setTrayTitle(video.paused ? "" : seriesTitle);
+  presenceData.details = seriesTitle;
```
Comment (Bas950) [inline websites/A/Aniorb/presence.ts:78]: just dont set the traytitle if its paused dont set it to ""

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/A/Aniorb/presence.ts [hunk 18]
```diff
@@ -64,17 +64,14 @@ else if (
...
-    );
+    timestamps = presence.getTimestampsfromMedia(video);
   presenceData.smallImageKey = video.paused ? "pause" : "play";
   presenceData.smallImageText = video.paused
     ? (await strings).pause
     : (await strings).play;
   presenceData.startTimestamp = timestamps[0];
   presenceData.endTimestamp = timestamps[1];
-  presence.setTrayTitle(video.paused ? "" : seriesTitle);
+  presence.setTrayTitle(video.paused);
```
[the remark this answers is not in the document]
Comment (ICEPrey) [reply websites/A/Aniorb/presence.ts]: bruh

File: websites/A/Aniorb/presence.ts [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+    seriesTitle = series.textContent.toUpperCase(),
+    episode = document.querySelector("#__next > div > div.flex.justify-between > div.w-full.justify-center.items-center.min-h-screen.lg\\:h-full.lg\\:w-10\\/12 > div > div.flex.flex-col.pb-2.xl\\:w-player.justify-between.items-center.w-full.text-white.my-4 > div.flex.w-full.justify-between.items-end > span").textContent,
+    timestamps = presence.getTimestampsfromMedia(video);
+  presenceData.smallImageKey = video.paused ? "pause" : "play";
+  presenceData.smallImageText = video.paused
+    ? (await strings).pause
+    : (await strings).play;
+  presenceData.startTimestamp = timestamps[0];
+  presenceData.endTimestamp = timestamps[1];
+  (!video.paused) presence.setTrayTitle(seriesTitle);
```
Comment (StrikerFRFX) [inline websites/A/Aniorb/presence.ts:74]: Isn't this supposed to be an if?

Comment (ICEPrey) [reply websites/A/Aniorb/presence.ts:74]: no?

**Review by StrikerFRFX — APPROVED** [review]

**Comment by ICEPrey** [comment]
> shit, didnt notice. codacy.

Ill fix it rn give me a second

File: websites/A/Aniorb/presence.ts [hunk 20]
```diff
@@ -0,0 +1,114 @@
...
+  presenceData.smallImageKey = "watching";
+} else if (
+  document
+    .querySelector(
+      "#__next > div > div.flex.justify-between > div.w-full.justify-center.items-center.min-h-screen.lg\\:h-full.lg\\:w-10\\/12 > div > div.flex.flex-col.pb-2.xl\\:w-player.justify-between.items-center.w-full.text-white.my-4 > div.w-full.py-4.uppercase.flex.flex-col.items-start.lg\\:items-start > span"
+    )
+    .textContent.toUpperCase() !== null &&
+  document.querySelector("video") !== null &&
+  video.video
+)
```
Comment (SlowLife1661) [inline websites/A/Aniorb/presence.ts:75]: Switch them

**Review by SlowLife1661 — CHANGES_REQUESTED** [review]

**Review by StrikerFRFX — APPROVED** [review]
Preferably if you could upload new proof of working since there were a lot of changes. Would be appreciated 👍

**Review by Bas950 — CHANGES_REQUESTED** [review]
please upload new proof of it working :)

**Comment by ICEPrey** [comment]
> please upload new proof of it working :)

The website is going to change its video player soon, so I am going to wait until that happens and then i fix the presence

**Pull request closed without merging**